File: models/notes.py
# coding: utf-8

import logging

from odoo import api, fields, models

_logger = logging.getLogger(__name__)

class Partner(models.Model):
    _name = 'res.partner'
    _inherit = 'res.partner'

    notes = fields.One2many('pos.customer.note', 'partner_id', 'Notes')

class Note(models.Model):
    _name = 'pos.customer.note'

    partner_id = fields.Many2one('res.partner', string='Customer', ondelete='cascade')
    text = fields.Text('Note')
    resolved = fields.Boolean('Resolved?', default=False)

    @api.model
    def create_from_ui(self, note):
        _logger.info("DOING SOME STUFF")
        _logger.info("note: %s", note)
        note_id = note.pop('id', False)
        if note_id:  # Modifying existing note
            self.browse(note_id).write(note)
        else:
            note_id = self.create(note).id
        return note_id

File: static/src/js/notes.js
odoo.define('pos.customer_notes', function (require) {
"use strict";

var models = require('point_of_sale.models');

// At POS Startup, load any unresolved customer notes, and add them to the relevant customer
models.load_models({
    model: 'pos.customer.note',
    fields: ['text','partner_id', 'resolved'],
    domain: function(self){ return [['resolved','=',false]]; },
    loaded: function(self,notes){
        var partner;
        var note;
        for (var i = 0; i < notes.length; i++){
            note = notes[i];
            partner = self.db.partner_by_id[note.partner_id[0]]
            partner.notes_by_id = partner.notes_by_id || {};
            partner.notes_by_id[note.id] = note;
        }
    },
});

});
